screens/Menu/CourseSchedule.js

```diff
@@ -23,53 +24,135 @@ import PropTypes from "prop-types";
+    const [selectedCalendarId, setSelectedCalendarId] = React.useState(null);
+    const [calendarEventsList, setCalendarEventsList] = React.useState(null);
+    const [switchVal, setSwitchVal] = React.useState("false");
+    const [refresh, setRefresh] = React.useState(false);
+
+    //TODO Show only events after current date
+    // let currentDate = new Date();
+
+    const getCalendarId = async () => {
+        let calendarId = await AsyncStorage.getItem("calendarId");
+        setSelectedCalendarId(calendarId);
+    };
+
+    const getAccessToken = async () => {
+        let AccessToken = await AsyncStorage.getItem("accessToken");
+        setAccessToken(AccessToken);
+    };
 
-     /**
+    const getSwitchValue = async () => {
+        let switchValue = await AsyncStorage.getItem('switchVal');
+        setSwitchVal(switchValue);
+    };
+
+    const getCalendarEvents = async () => {
+        let calendarEvents = await fetch('https://www.googleapis.com/calendar/v3/calendars/' + selectedCalendarId + '/events', {
+            headers: { Authorization: `Bearer ${accessToken}` }
+        });
+        let resp = await calendarEvents.json();
+        setCalendarEventsList(getFilteredGoogleCalendarEvents(resp));
+        setRefresh(false);
+    }
+
+    const getFilteredGoogleCalendarEvents = (resp) => {
+        var filteredList = resp.items.map(element => {
+            return { id: element.id, summary: element.summary, description: element.description, location: element.location };
+        });
+        return filteredList;
+    };
+
+    const handleRefresh = () => {
+        setRefresh(true);
+        getCalendarEvents();
+    }
+
+    const CourseScheduleScreen = true;
+
+    /**
      * The method will slide the side menu from the right side of the screen
      * @param  {} =>{props.navigation.openDrawer(
-     */
+    */
     const goToMenu = () => {
         props.navigation.openDrawer();
     };
 
-    
-    /**
-     * The method will let us navigate to the CourseSchedule screen
-     * @param  {} =>{props.navigation.navigate("CourseScheduleDetails"
-     */
-    const goToCourseSchedule = () => {
-        props.navigation.navigate("CourseScheduleDetails");
-    };
+    const goToDoubleSearch = (item) => {
+        props.navigation.navigate("DoubleSearch", { CourseScheduleScreen: CourseScheduleScreen, CourseScheduleLocation: item.location });
+    }
 
-    // TODO: Static data for now just to layout the SectionList 
-    const data = [{
-        title: "Wed, Feb 5",
-        data: ["SOEN 357 Lecture", "SOEN 357 Tutorial", "SOEN 345 Lecture"
-        ]
-    }];
+    useEffect(() => {
+        const intervalId = setInterval(() => {
+            getCalendarId();
+            getAccessToken();
+            getSwitchValue();
+        }, 1);
+        getCalendarEvents();
+        return () => clearInterval(intervalId);
+    }, [selectedCalendarId, accessToken]);
 
+    if (switchVal == "true") {
```

Comment:
Extend this if statement with `calendarEventsList`   : `if (switchVal == "true" && calendarEventsList)`
This is to match the inner condition at line `106`.

Reply:
Got it !!